xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -521,15 +521,17 @@ bool CDVDDemuxFFmpeg::Open(const std::shared_ptr<CDVDInputStream>& pInput, bool
   bool isBluray = pInput->IsStreamType(DVDSTREAM_TYPE_BLURAY);
   // don't re-open mpegts streams with hevc encoding as the params are not correctly detected again
   if (iformat && (strcmp(iformat->name, "mpegts") == 0) && !fileinfo && !isBluray &&
+      m_pFormatContext->nb_streams > 0 && m_pFormatContext->streams &&
```

Comment:
checking `m_pFormatContext->streams (!= nullptr)` could be sufficient. but better safe than sorry i guess

Reply:
It should be an error condition in external code that causes nb_streams > 0 and streams == nullptr. I suggest you fix the external code. But given the error could be introduced again, it's probably best to be defensive and do both checks here.

Reply:
Yeah the two added conditions is a bit nonsense...
Maybe we should just assert (or log) before and use nb_streams > 0 in the condition?
It comes down to the fact we want Kodi crashing or not in the hypothetical case there is some logic error in depends...

Reply:
Good point, I didn't realize `nb_streams` was from an ffmpeg call (though the snake case should have tipped me off).

Yes, guard against errant depends. I'll approve the PR.

BTW, I prefer being explicit and type-safe, so I use `!= nullptr` instead of casting a pointer to bool.

Reply:
i thought m_pFormatContext->streams might be int unless i digged deeper. +1 for being explicit